spec/controllers/gestionnaires/groupe_gestionnaires_controller_spec.rb

```diff
@@ -110,4 +110,27 @@
       it { expect(response).to redirect_to(gestionnaire_groupe_gestionnaires_path) }
     end
   end
+
+  describe "#tree_structure" do
+    let!(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+    subject { get :tree_structure, params: { id: groupe_gestionnaire.id } }
+
+    context "when not logged" do
+      before { subject }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context "when logged in" do
+      let!(:child_groupe_gestionnaire) { create(:groupe_gestionnaire, ancestry: "/#{groupe_gestionnaire.id}/", gestionnaires: []) }
+      before do
+        sign_in(gestionnaire.user)
+        subject
+      end
+
+      it do
+        expect(response).to have_http_status(:ok)
```

Comment:
comme une PR précédente: déplacer le `subject` dans le `it` pour faciliter la maintenance des tests